primefaces/src/main/java/org/primefaces/model/filter/ComparableFilterConstraint.java

```diff
@@ -23,17 +23,52 @@
  */
 package org.primefaces.model.filter;
 
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.text.ParseException;
 import javax.faces.context.FacesContext;
 import java.util.Locale;
+import java.util.function.BiPredicate;
 
-public abstract class ComparableFilterConstraint extends StringFilterConstraint {
+public abstract class ComparableFilterConstraint implements FilterConstraint {
 
     @Override
     public boolean isMatching(FacesContext ctxt, Object value, Object filter, Locale locale) {
-        if (!(value instanceof Comparable) || !(filter instanceof Comparable)) {
-            throw new IllegalArgumentException("Invalid type: " + value.getClass() + ". Valid type: " + Comparable.class.getName());
+        if (value == null || filter == null) {
+            return false;
         }
+        if (value instanceof Number) {
+            BigDecimal filterBigDecimal = toBigDecimal(filter, locale);
+            return filterBigDecimal != null && getPredicate().test(toBigDecimal(value, locale), filterBigDecimal);
+        }
+        else {
+            return getPredicate()
+                    .test(StringFilterConstraint.toString(value, locale),
+                            StringFilterConstraint.toString(filter, locale));
+        }
+    }
 
-        return super.isMatching(ctxt, value, filter, locale);
+    static BigDecimal toBigDecimal(Object object, Locale locale) {
+        if (object instanceof BigDecimal) {
+            return (BigDecimal) object;
+        }
+        if (object instanceof Number) {
+            return new BigDecimal(object.toString());
```

Comment:
Can object be NULL here which would cause an NPE on the `toString() call?

Reply:
No, it has been checked on line 38 and instanceof null is false.